### ISSUE TITLE
Remember angular size of regions used with the FluxProfileEstimator

### ISSUE THREAD

**Issue by LauraOlivera** [gammapy/gammapy#3863]
When using the `FluxProfileEstimator` the number of regions used (i.e. their size) will obviously impact the result. However, in the resulting `FluxPoints` object, only the projected distance information is kept. Meaning that this information (the solid angle size of the regions used) is lost.

This is inconvenient when one wants to, for example, compare the flux profile to the prediction of a model, because the exact regions used to create the `FluxProfileEstimator` need to be also saved alongside the profile in order to correct for the region sizes. The prediction of the model can typically be derived in smaller spatial bins to better characterize the shape, but then has to be corrected to match the flux level of the points derived from data.

**Describe the solution you'd like**
A solution could be to maybe add another `Map` to the container. For example,  `FluxPoints.angular_size` that is 0 or nans when dealing with "normal" flux points, and that contains the region size in sr for the flux profiles. The regions resulting of `make_orthogonal_rectangle_sky_regions` all have (almost exactly) the same sizes, so maybe just one number would work as well. But for generality maybe it is better to keep them all.

**Describe alternatives you've considered**
Of course another option is to save the list of regions, or directly their sizes alongside the `FluxPoints` and handle them together every time, but I would argue it is quite unpractical in terms of bookkeeping, and it would be nicer if the flux points container remembered this information.

**Comment by adonath** [comment]
Thanks a lot @LauraOlivera, I fully agree this is important information to keep.

I think there is already a very good place to store this kind of information: the `RegionGeom` itself, which can directly store the `Region` objects. It then also supports plotting, computing solid angles, serialization etc. What is currently missing is the possibility to "assign" regions to bins in a `MapAxis`. Effectively it is the same problem as handling an energy dependent regions size. In one case the region varies with energy, in the second case in varies with label, bin number of maybe projected distance (this remains to be decided). A few month ago in the context for the energy dependent region size, I worked on a little prototype for this. To provide a possible starting point and base for further discussion I'll link to it here again:
- Branch: https://github.com/adonath/gammapy/tree/region_geom_array
- Notebook: https://github.com/adonath/notebooks-public/blob/master/region-geom-array.ipynb

As a very quick, low effort solution I think one could even just store the list of regions as a `CompoundRegion` on the `RegionGeom` object. This will miss the direct mapping of the axis bin to the region, but implicitly it would work by "flattening" the compound region to a list of regions, which has the same length and order as the corresponding axis. This would at least keep the information, even if there is now convenient access to it...

**Comment by adonath** [comment]
The base of my proposal above was a `RegionArray` class, essentially and extension of the `Region` class in Astropy's `regions` package to support arrays of positions, region sizes etc. I asked in the Slack channel whether array support for `Region` objects is foreseen in future and the answer was that this is rather the scope of `photutils` and the corresponding apertures classes. Not sure if this is maybe relevant here. 

Given my prototype above the kind of information such as region size / positions would be stored on the `RegionArray`, and be available on the `FluxPoints` object on each data member like:
```
fp = FluxPoints()
fp.flux.geom.region.positions # returns SkyCoord([pos_1, pos_2, pos_3, ...])
fp.flux.geom.region.radii # returns Quantity([r_1, r_2, r_2, ])
```

**Comment by LauraOlivera** [comment]
Thanks a lot @adonath for your response! Yes, I agree that the `RegionGeom` is perfect, and I had actually forgotten about the `CompoundRegion`, which I think is a good, zero-th order solution to the problem at hand. I think that if this was documented in the `FluxProfileEstimator` page, it would be an acceptable solution.

Of course the second option you propose is nicer, because it solves other problems too (energy dependent region sizes). If I understand it right, one would need to create two specific `RegionArray` (sky and pixel) for each shape, right? Which I guess is not too extreme since we don't support that many shapes...

**Comment by adonath** [comment]
> If I understand it right, one would need to create two specific RegionArray (sky and pixel) for each shape, right? Which I guess is not too extreme since we don't support that many shapes...

Yes, and even this is only needed, because `regions` does not support arrays. The important feature from the array support is actually the broadcasting, which simplifies the implementation a lot. Otherwise one could even work with a simple list of regions, however an alternative solution could be a class like:

```python3
class MapAxisRegion(Region):
    """Map axis"""
    def __init__(self, regions, axis):
        assert len(regions) == axis.nbin
        self.regions = regions
        self.axis = axis

    def contains(self, wcs):
        # mimic region API here and loop over regions and broadcast...
        pass

    def to_pixel(self, wcs):
        # mimic region API here and loop over regions and broadcast...
        return MapAxisPixelRegion(pix_regions, axis=self.axis)

    def to_artist(self, **kwargs):
        pass


        
 

```
